### PR TITLE
Consul: fix build on Linux

### PR DESCRIPTION
**Pull request by DoomHammer**
- [X] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [X] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [X] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

-----

Fixes #565

### CODE REVIEW

**Comment by DoomHammer** [comment]
Idea taken shamelessly from https://github.com/Homebrew/homebrew-core/pull/2032/files

File: Formula/gnupg2.rb [hunk 1]
```diff
@@ -56,6 +56,7 @@ def install
       --sbindir=#{bin}
       --enable-symcryptrun
       --disable-agent
+      --disable-nls # Due to https://github.com/Linuxbrew/homebrew-core/issues/311
```
Comment (rwhogg) [inline Formula/gnupg2.rb]: This was added accidentally, right?

**Comment by sjackman** [comment]
Does Homebrew/core need this patch as well?

**Comment by DoomHammer** [comment]
Cannot test myself but would say upstream needs it.

File: Formula/gnupg2.rb (hunk 1 above)
Comment (DoomHammer) [inline Formula/gnupg2.rb]: Totally right :)

**Comment by DoomHammer** [comment]
Should I submit it to Homebrew as well?

**Comment by sjackman** [comment]
I wasn't able to reproduce any build error of `consul` on Mac OS 10.11.5.
https://gist.github.com/dfbd0622bdc6bc599e3b3e731291ea38

**Comment by DoomHammer** [comment]
Strange. Is `gox` somehow available on Mac OS during installation? Is there a way for me to get an SSH access to a Mac machine to test it properly?

**Comment by sjackman** [comment]
I'm not sure. Sorry. Ask the consul developers perhaps?

**Comment by DoomHammer** [comment]
I see now the problem is not with Consul itself but rather one of its deps. Will dig further.

**Comment by DoomHammer** [comment]
Nope, that one was false. Most likely my previous "fix" for Linuxbrew is now causing us pain somewhere down below...

**Comment by DoomHammer** [comment]
@sjackman are you OK with this solution?

**Comment by sjackman** [comment]
Sure, if it makes it work, looks fine to me. Please add `unless OS.mac?` or `if OS.linux?` as appropriate.

**Comment by DoomHammer** [comment]
Added `if OS.linux?` as the original `make` seems broken only here and no idea why (Google shows no relevant results searching for the error and that seems scary).

**Comment by DoomHammer** [comment]
Entirely different approach waiting for new upstream release.

**Comment by DoomHammer** [comment]
How do you feel about it @sjackman ?

**Comment by DoomHammer** [comment]
Ehm, @sjackman, @rwhogg , any comments?

**Comment by sjackman** [comment]
Sorry I've been AWOL. See my comment. Otherwise looks good to me.

**Comment by DoomHammer** [comment]
Mac uses `make`. We used `make dev` as a result of my previous PR. The root cause is that `gox` builds binaries for every platform imaginable (from Go's perspective) and there is a weird bug for one of the archs on Linux host. Since `darwin/arm` binary is not necessary on Linux anyway my new approach was to comment the platform from the build script.

**Comment by sjackman** [comment]
:+1: Got it. Fine to merge by me.

**Pull request closed without merging**